pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2279,16 +2279,49 @@ type BasicAuth struct {
 	PasswordSecret *apiv1.SecretKeySelector `json:"passwordSecret,omitempty" protobuf:"bytes,2,opt,name=passwordSecret"`
 }
 
+// ClientCertAuth holds necessary information for client authentication via certificates
+type ClientCertAuth struct {
+	ClientCertSecret *apiv1.SecretKeySelector `json:"clientCertSecret,omitempty" protobuf:"bytes,1,opt,name=clientCertSecret"`
```

Comment:
I'm not sure we suffix `Secret` in other places, can you check please?

Reply:
yes, its done e.g. here: https://github.com/argoproj/argo-workflows/blob/1a7d50fd32a30070e78bdf544dd709c20647378d/pkg/apis/workflow/v1alpha1/workflow_types.go#L2032-L2036